libensemble/manager.py

```diff
@@ -398,6 +433,7 @@ def _update_state_on_worker_msg(self, persis_info, D_recv, w):
             if w in self.persis_pending:
                 self.persis_pending.remove(w)
                 self.W[w-1]['active'] = 0
+            self._freeup_resources(w)  # SH TODO - check persistent pending does not screw this up
```

Comment:
Is this being checked somewhere, or are we not worrying about it for now?

Reply:
Good catch, this is an old comment. I'm fixing now.